documentation/src/main/asciidoc/reference/mapper-orm-bridge-routingbridge.asciidoc

```diff
@@ -0,0 +1,239 @@
+[[mapper-orm-bridge-routingbridge]]
+= Routing bridges
+// Legacy anchor from earlier 6.0 beta versions
+[[mapper-orm-bridge-routingkeybridge]]
+
+== Basics
+
+A routing bridge is a pluggable component that defines, at runtime,
+whether an entity should be indexed and <<concepts-sharding-routing,to which shard the corresponding indexed document should be routed>>.
+It is applied to an indexed entity type with the `@Indexed` annotation,
+using its `routingBinder` attribute (`@Indexed(routingBinder = ...)`).
+
+Implementing a routing bridge requires two components:
+
+. A custom implementation of `RoutingBinder`, to bind the bridge to an indexed entity type at bootstrap.
+This involves declaring the properties of the indexed entity type that will be used by the routing bridge
+and instantiating the routing bridge.
+. A custom implementation of `RoutingBridge`, to route entities to the index at runtime.
+This involves extracting data from an instance of the type, transforming the data if necessary,
+and defining the current route (or marking the entity as "not indexed").
++
+If routing can change during the lifetime of an entity instance,
+you will also need to define the potential previous routes,
+so that Hibernate Search can find and delete previous documents indexed for this entity instance.
+
+In the sections below, you will find examples for the main use cases:
+
+* <<mapper-orm-bridge-routingbridge-conditionalindexing>>
+* <<mapper-orm-bridge-routingbridge-routingkey>>
+
+[[mapper-orm-bridge-routingbridge-conditionalindexing]]
+== Using a routing bridge for conditional indexing
+// Search 5 anchors backward compatibility
+[[example-search-mapping-indexinginterceptor-blog]]
+
+Below is a first example of a custom routing bridge that
+disables indexing for instances of the `Book` class if their status is `ARCHIVED`.
+
+.Implementing and using a `RoutingBridge` for conditional indexing
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/mapper/orm/bridge/routingbridge/conditionalindexing/BookStatusRoutingBinder.java[tags=binder]
+----
+<1> The binder must implement the `RoutingBinder` interface.
+<2> Implement the `bind` method in the binder.
+<3> Declare the dependencies of the bridge,
+i.e. the parts of the entity instances that the bridge will actually use.
+See <<mapper-orm-bridge-bridgedelement-dependencies>>
+for more information about declaring dependencies.
+<4> Call `context.bridge(...)` to define the routing bridge to use.
+<5> Pass the expected type of indexed entities.
```

Comment:
We already know what the actual indexed type is, that's right. Here we need the user to pass the expected type along with the bridge so that we can check the expected type is a super type of the actual indexed type.
Thanks to how the method in define, if that condition is met, we're sure we can apply the bridge to the actual indexed type. That's pretty much the only reason we ask users to pass a type.

We already expose a similar API for the `ValueBridge`, by the way.